File: Programming Fundamentals/Programming-Fundamentals-September-2017/DictionariesLambdaAndLINQ-Exercise/01.Phonebook/Phonebook.cs
namespace _01.Phonebook
{
    using System;
    using System.Collections.Generic;
    using System.Linq;
    public class Phonebook
    {
        public static void Main()
        {
            var phonebook = new Dictionary<string, string>();

            string input = Console.ReadLine();

            while (input != "END")
            {
                var commandArguments = input.Split(' ').ToList();

                string command = commandArguments[0];
                
                if (command == "A")
                {
                    string name = commandArguments[1];
                    string telephone = commandArguments[2];

                    if (!phonebook.ContainsKey(name))
                    {
                        phonebook.Add(name, telephone);
                    } else
                    {
                        phonebook[name] = telephone;
                    }
                }
                else
                {
                    string name = commandArguments[1];

                    if (phonebook.ContainsKey(name))
                    {
                        Console.WriteLine($"{name} -> {phonebook[name]}");
                    }
                   else
                    {
                        Console.WriteLine($"Contact {name} does not exist.");
                    }
                }
                input = Console.ReadLine();
            }
        }
    }
}

File: Software Technologies/Java Basic Syntax Exercises/src/Phonebook.java
import java.util.HashMap;
import java.util.LinkedHashMap;
import java.util.Map;
import java.util.Scanner;

public class Phonebook {
    public static void main(String[] args) {

        Scanner scanner = new Scanner(System.in);
        String input;
        LinkedHashMap<String, String> phonebook = new LinkedHashMap<>();

        while (!"END".equals(input = scanner.nextLine())) {

            String[] tokens = input.split(" ");

            String command = tokens[0];

            switch (command) {
                case "A":
                    String name = tokens[1];
                    String number = tokens[2];

                    phonebook.put(name, number);
                    break;
                case "S":
                    name = tokens[1];
                    if (phonebook.containsKey(name)) {
                        String foundNumber = phonebook.get(name);
                        System.out.printf("%s -> %s%n",name, foundNumber );
                    } else {
                        System.out.printf("Contact %s does not exist. %n", name);
                    }
                    break;
            }
        }
    }
}
